bliss/metrics.py

```diff
@@ -40,50 +43,60 @@ def __init__(
             total_true_n_sources: Total number of true sources over batches seen.
-            total_correct_class: Total # of correct classifications over matched objects.
-            total_n_matches: Total # of matches over batches.
-            conf_matrix: Confusion matrix (galaxy vs star)
             disable_bar: Whether to show progress bar
+            gal_tp: true positives = # of sources correctly classified as a galaxy
+            gal_fp: false positives = # of sources incorrectly classified as a galaxy
+            gal_fn: false negatives = # of sources incorrectly classified as a star
+            gal_tn: true negatives = # of sources correctly classified as a star
         """
         super().__init__(dist_sync_on_step=dist_sync_on_step)
 
         self.slack = slack
         self.disable_bar = disable_bar
 
         self.add_state("tp", default=torch.tensor(0), dist_reduce_fx="sum")
-        self.add_state("tp_gal", default=torch.tensor(0), dist_reduce_fx="sum")
         self.add_state("fp", default=torch.tensor(0), dist_reduce_fx="sum")
         self.add_state("avg_distance", default=torch.tensor(0.0), dist_reduce_fx="sum")
         self.add_state("total_true_n_sources", default=torch.tensor(0), dist_reduce_fx="sum")
-        self.add_state("total_correct_class", default=torch.tensor(0), dist_reduce_fx="sum")
-        self.add_state("conf_matrix", default=torch.tensor([[0, 0], [0, 0]]), dist_reduce_fx="sum")
+        self.add_state("gal_tp", default=torch.tensor(0), dist_reduce_fx="sum")
+        self.add_state("gal_fp", default=torch.tensor(0), dist_reduce_fx="sum")
+        self.add_state("gal_fn", default=torch.tensor(0), dist_reduce_fx="sum")
+        self.add_state("gal_tn", default=torch.tensor(0), dist_reduce_fx="sum")
 
     # pylint: disable=no-member
     def update(self, true: FullCatalog, est: FullCatalog) -> None:  # type: ignore
-        """Update the internal state of the metric including tp, fp, total_true_n_sources, etc."""
+        """Update the internal state of metrics including tp, fp, total_coadd_gal_matches, etc."""
         assert true.batch_size == est.batch_size
 
         count = 0
-        desc = "Detection Metric per batch"
+        desc = "Bliss Metric per batch"
         for b in tqdm(range(true.batch_size), desc=desc, disable=self.disable_bar):
             ntrue, nest = true.n_sources[b].int().item(), est.n_sources[b].int().item()
             tlocs, elocs = true.plocs[b], est.plocs[b]
+            tgbool, egbool = true["galaxy_bools"][b].reshape(-1), est["galaxy_bools"][b].reshape(-1)
             if ntrue > 0 and nest > 0:
                 mtrue, mest, dkeep, avg_distance = match_by_locs(tlocs, elocs, self.slack)
-                tp = len(elocs[mest][dkeep])  # n_matches
-                true_galaxy_bools = true["galaxy_bools"][b][mtrue][dkeep]
-                tp_gal = true_galaxy_bools.bool().sum()
+                tp = len(elocs[mest][dkeep])
```

Comment:
Is the `if` statement above OK? It seems like if, say, `ntrue == 0`, we could still have some false positives, for example

Reply:
Right. The if-statement was pre-existing but I didn't think of that case - will fix.